### PR TITLE
Add broken Xref checker

### PR DESCRIPTION
**Pull request by settermjd**
This PR adds a custom Antora generator script that can find and report broken Xrefs in the documentation and updates the docs to show how to use it.

### CODE REVIEW

File: docs/build-the-docs.md [hunk 1]
```diff
@@ -35,6 +35,33 @@ antora site.yml
...
+There are a host of ways to check for broken links, including NPM's broken link checker.
+However, this custom generator, provided by the core Antora team, is able to check [the native Antora xref links](https://docs.antora.org/antora/1.0/asciidoc/page-to-page-xref/#xref-and-page-id-anatomy).
+It doesn't check external links, but it's still a good start.
+To use it, you need to pass the custom generator (in `./generator/xref-validator`) to the `generate` command, as in the example below.
+```console
+antora generate \
+    --pull \
+    --ui-bundle-url https://github.com/owncloud/docs-ui/releases/download/1.1.0/ui-bundle.zip \
+    --generator=./generator/xref-validator \
+    site.yml
```
Comment (mmattel) [inline docs/build-the-docs.md]: ``site.yml`` --> ``site.prod.yml``

File: docs/build-the-docs.md [hunk 2]
```diff
@@ -35,6 +35,33 @@ antora site.yml
...
+#### Testing For Broken Xref Links
+
+There are a host of ways to check for broken links, including NPM's broken link checker.
+However, this custom generator, provided by the core Antora team, is able to check [the native Antora xref links](https://docs.antora.org/antora/1.0/asciidoc/page-to-page-xref/#xref-and-page-id-anatomy).
+It doesn't check external links, but it's still a good start.
+To use it, you need to pass the custom generator (in `./generator/xref-validator`) to the `generate` command, as in the example below.
+```console
+antora generate \
+    --pull \
+    --ui-bundle-url https://github.com/owncloud/docs-ui/releases/download/1.1.0/ui-bundle.zip \
```
Comment (mmattel) [inline docs/build-the-docs.md]: I would reference to the local ``ui-bundle.zip`` not the one from docs/releases

**Comment by mmattel** [comment]
**First**
I used that branch to test, but I get the following error 

``error: Generator not found or failed to load.``

with option ``--stack-trace``:
``Error: Cannot find module '@antora/content-aggregator'``

**Second**
I created https://github.com/owncloud/docs/pull/205 (Have an own page for a broken link checker)
Why not adding it there ? Would make more sense to me as the other once can also check external links.

File: docs/build-the-docs.md (hunk 1 above)
Comment (settermjd) [reply docs/build-the-docs.md]: That would have to happen later, as that file's not part of this branch. The docs can be updated later.

File: docs/build-the-docs.md (hunk 2 above)
Comment (settermjd) [reply docs/build-the-docs.md]: I believe the release artifact is the right one, for the time being, until the build pipeline is finalised.

Comment (mmattel) [reply docs/build-the-docs.md]: sorry disagree. we currently do have regular updates in docs-ui and as long the automated process is not in place, any reference to an old bundle will counteract the updates made. just my 2c

**Comment by settermjd** [comment]
That error shouldn't appear when the Antora packages are installed locally. As #216 is merged to master, and rebased now onto this PR, it should be fine to merge this now. I'll update the README to cover the inclusion of package.json and package-lock.json in a separate PR.

**Pull request merged**